python/sglang/srt/layers/sampler.py

```diff
@@ -43,7 +43,9 @@ def forward(
                 torch.isnan(probs), torch.full_like(probs, 1e-10), probs
             )
 
-        if global_server_args_dict["sampling_backend"] == "flashinfer":
+        if torch.all(sampling_info.top_ks == 1):
```

Comment:
add some comments? curious does this cover the case for both top_k=1 and temp=0? i guess so given the test is provided temp=0

Reply:
Yes, in internally, temperature=0 will set top_k=1